internal/migrations/observability/api/convert.go

```diff
@@ -0,0 +1,112 @@
+	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	obsruntime "github.com/openshift/cluster-logging-operator/internal/runtime/observability"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConvertLoggingToObservability converts a logging.ClusterLogging and/or logging.ClusterLogForwarder
+// to observability.ClusterLogForwarder
+func ConvertLoggingToObservability(k8sClient client.Client, loggingCL *logging.ClusterLogging, loggingCLF *logging.ClusterLogForwarder, outputSecrets map[string]*corev1.Secret) *obs.ClusterLogForwarder {
+	// Legacy Workflow
+	if loggingCL != nil && loggingCL.Name == constants.SingletonName && loggingCL.Namespace == constants.OpenshiftNS {
+		clToObsCLF := obsruntime.NewClusterLogForwarder(loggingCL.Namespace, loggingCL.Name, runtime.Initialize)
+		// Legacy ClusterLogging only
+		if loggingCLF == nil {
+			obsClfSpec := convertLegacyClusterLogging(&loggingCL.Spec)
+			clToObsCLF.Spec = *obsClfSpec
+			return clToObsCLF
+		}
+		// Legacy ClusterLogging + ClusterLogForwarder
+		clToObsCLF.Spec = convertClusterLogForwarder(loggingCL, loggingCLF, outputSecrets, true)
+		return clToObsCLF
+	}
+
+	// custom clf
+	obsCLF := obsruntime.NewClusterLogForwarder(loggingCLF.Namespace, loggingCLF.Name, runtime.Initialize)
+	obsCLF.Spec = convertClusterLogForwarder(loggingCL, loggingCLF, outputSecrets, false)
+
+	return obsCLF
+}
+
+// convertLegacyClusterLogging generates the output and pipelines determined by the logstore type of the ClusterLoggingInstance
+func convertLegacyClusterLogging(loggingCLSpec *logging.ClusterLoggingSpec) *obs.ClusterLogForwarderSpec {
+	obsDefaultOut := *generateDefaultOutput(loggingCLSpec.LogStore)
+
+	var obsCollSpecs *obs.CollectorSpec
+	if loggingCLSpec.Collection != nil {
+		obsCollSpecs = &obs.CollectorSpec{
+			Resources:    loggingCLSpec.Collection.Resources,
+			NodeSelector: loggingCLSpec.Collection.NodeSelector,
+			Tolerations:  loggingCLSpec.Collection.Tolerations,
+		}
+	}
+
+	return &obs.ClusterLogForwarderSpec{
+		ServiceAccount: corev1.LocalObjectReference{
+			Name: constants.CollectorServiceAccountName,
+		},
+		Outputs: []obs.OutputSpec{
+			obsDefaultOut,
+		},
+		Pipelines: []obs.PipelineSpec{
+			{
+				Name:       obsDefaultOut.Name + "-pipeline",
+				InputRefs:  []string{string(obs.InputTypeApplication), string(obs.InputTypeInfrastructure)},
+				OutputRefs: []string{obsDefaultOut.Name},
+			},
+		},
+		Collector: obsCollSpecs,
+	}
+}
+
+func convertClusterLogForwarder(loggingCl *logging.ClusterLogging, loggingClf *logging.ClusterLogForwarder, secrets map[string]*corev1.Secret, isLegacyDeployment bool) obs.ClusterLogForwarderSpec {
+	var logStoreSpec *logging.LogStoreSpec
+	saName := constants.CollectorServiceAccountName
```

Comment:
I think this will be needed but what are the implications for resource cleanup?

Reply:
In multiCLF and 6.0, users provide a SA so it doesn't have an ownerref. They are responsible for cleanup.  The difference here is we are essentially "transferring" ownership to the admin and its no longer cleaned up or managed by the operator.  Maybe we need a docs note of "whats changed" but otherwise I do not think there is cleanup